packages/config/index.js

```diff
@@ -4,34 +4,68 @@ var path = require('path');
 
 var root = require('pkg-dir').sync(process.cwd());
 
-var npmConfig = require('./lib/parse-env')('hops');
-if (!Object.keys(npmConfig).length) {
-  try {
-    require.resolve(path.join(root, 'package.json'));
-    npmConfig = require(path.join(root, 'package.json')).config.hops;
-  } catch (_) {}
+var explorer = require('cosmiconfig')('hops', {
+  rcExtensions: true,
+  stopDir: root,
+  sync: true,
+});
+
+function getRawConfig() {
+  var result = explorer.load(process.cwd());
+  return Object.assign(
+    {
+      https: false,
+      host: '0.0.0.0',
+      port: 8080,
+      locations: [],
+      basePath: '',
+      assetPath: '',
+      browsers: '> 1%, last 2 versions, Firefox ESR',
+      node: 'current',
+      envVars: { HOPS_MODE: 'dynamic' },
+      moduleDirs: [],
+      appDir: '.',
+      buildDir: 'build',
+      cacheDir: 'node_modules/.cache/hops',
+    },
+    result ? result.config : {}
+  );
 }
 
-function extendConfig(config) {
-  if (npmConfig.extends) {
+function applyExtension(config) {
+  var result = Object.assign({}, config);
+  if (config.extends) {
     try {
-      require.resolve(npmConfig.extends);
-      Object.assign(config, require(npmConfig.extends));
+      require.resolve(config.extends);
+      Object.assign(result, require(config.extends));
     } catch (_) {
-      Object.assign(config, require(path.join(root, npmConfig.extends)));
+      Object.assign(result, require(path.join(root, config.extends)));
```

Comment:
In the past we've sometimes run into problems here, mainly when the `hops-config` is installed in the wrong folder or something similar.
Would it make sense to `try/catch` the second require too and print a friendly warning?

Reply:
We could also use `resolve-cwd` or one of its related projects: https://www.npmjs.com/package/resolve-cwd#related